apps/omg_watcher/lib/omg_watcher/exit_processor.ex

```diff
@@ -44,6 +45,7 @@ defmodule OMG.Watcher.ExitProcessor do
   Accepts events and processes them in the state - new exits are tracked.
   Returns `db_updates`
   """
+  @decorate transaction_event(:ExitProcessor)
```

Comment:
Under which transaction are these transaction events collected?

Reply:
should be driven by `EthereumEventListeners` who are effectively managing the background jobs here, and calling into here.

Come to think of it, the above is quite secret knowledge. It would be cool to have sth like
```
@decorate transaction_or_event(...)
```

But otherwise, I think the way to go now would just be to `event` **everywhere** except where we're sure we're either handling an incoming request (`web` namespace) or acting on our own by starting a background job (`background` namespace)

Reply:
I'm gonna list why I don't like these annotations:
- they seem to couple us with appsignal 
- we don't know what the performance impact of decorations is and it seems like there's no way to turn them off (which is scary!)
- there ought to be a cleaner way to gather and measure events (Telemetry maybe?) and have a pluggable backend for metrics

Reply:
@InoMurko

>I'm gonna list why I don't like these annotations:

Generally, I agree with your 3 concerns here, but wouldn't we rather just get AppSignal metrics up and running quickly, and possibly move to something more elegant at a later stage?

I've just given it a quick scan, and it seems going the Telemetry way (+ AppSignal's custom metrics, to have the similar final effect, correct?), would require us to collect all the low level measurement values manually.

BTW, you suggested instrumenting using decorators in the first place, which part of the execution specifically made you change your opinion?

Regardless, I'll let you guys (@pthomalla) figure out the best first approach here.

Reply:
we could make our `decorate`, where we can turn off some measurement